Project updates and cleanups

friederbluemle opened the pull request:
A ton of project updates and cleanups. It's probably best to look at the individual commits. Let me know if you have any questions. Great library by the way!

shivasurya commented:
first of all thanks for spending your time to contribute.
btw,please give me little time till weekend to check the commits and get to you back :) 
awesome job :+1:

friederbluemle commented:
@shivasurya Thanks, sounds good. Let me know if there are any questions.
Version 2.1.0-alpha5 of the Android Gradle plug-in is now available. I amended my previous commit (alpha4) and updated my branch. Everything is at the latest again.

friederbluemle commented:
@shivasurya Would it help with the review if I move some of the commits over to different PR?
The commits are in a logical chronological order, so if you review them one-by-one, it should be pretty quick. Of course you don't really need to look at the two big commits that just remove unnecessary files from the repo (IDE files/folders and build output).

friederbluemle commented:
A few more upstream updates happened (Gradle `2.12` -> `2.13`; Gradle plug-in `2.1.0-alpha5` -> `2.1.0`). I used fixup commits to amend the previously committed changes.
@shivasurya Did you get a chance to review?

friederbluemle commented:
Hi @shivasurya, hope all is well. It's been a while, and I just stumbled upon this PR from last year :) I updated everything to the very latest stable dependency versions, and also squashed a couple of commits.

- Gradle wrapper `3.4.1`
- Android Gradle plugin `2.3.0`
- Build tools to `25.0.2`
- Support libs `25.3.1`
- Compile and target sdk `25`

friederbluemle commented:
Updated again to the latest

- Gradle wrapper `3.5`
- Android Gradle plugin `2.3.1`
